src/utils/random.cpp

```diff
@@ -89,8 +119,16 @@ bool save_rng_to_checkpoint_shared(persist& p, const lbann_comm* comm) {
     rank_in_world = std::to_string(El::mpi::Rank(El::mpi::COMM_WORLD));
   } else {
     rank_in_world = std::to_string(comm->get_rank_in_world());
-
   }
+
```

Comment:
This will not work right, because it saves only the state for a single thread. I think we should raise a warning until it is fixed properly, since otherwise this will confuse people.